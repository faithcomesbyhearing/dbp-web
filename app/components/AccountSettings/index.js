/**
 *
 * AccountSettings
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import SvgWrapper from '../SvgWrapper';
import PopupMessage from '../PopupMessage';
import messages from './messages';

class AccountSettings extends React.PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			email: this.props.profile.email,
			nickname: this.props.profile.nickname,
			name: this.props.profile.name,
			avatar: this.props.profile.avatar,
			popupOpen: false,
		};
	}

	handleEmailChange = (e) => {
		this.setState({ email: e.target.value });
	};

	render() {
		const { logout, profile } = this.props;
		const { email, popupOpen } = this.state;

		return (
			<div className="account-settings">
				<div
					role="button"
					tabIndex={0}
					onClick={logout}
					className="logout-button"
				>
					<FormattedMessage {...messages.logout} />
				</div>
				<section className="personal-info">
					<SvgWrapper
						className={'avatar-placeholder'}
						svgid={'avatar_placeholder'}
					/>
					{profile.nickname && profile.nickname !== 'undefined' ? (
						<h3 className="name">{profile.nickname}</h3>
					) : null}
					{profile.name && profile.name !== 'undefined' ? (
						<span className="name">{profile.name}</span>
					) : null}
				</section>
				<div className="email-section">
					<span className="title">e-mail</span>
					<span className="wrapper">
						<SvgWrapper
							className="icon"
							height="26px"
							width="26px"
							svgid="e-mail"
						/>
						<input
							onChange={this.handleEmailChange}
							placeholder="emailaddress@mail.com"
							value={email}
						/>
					</span>
				</div>
				{/* <DonateButton /> */}
				{popupOpen ? (
					<PopupMessage
						message={'Please select a valid image file.'}
						x={160}
						y={250}
					/>
				) : null}
			</div>
		);
	}
}

AccountSettings.propTypes = {
	logout: PropTypes.func,
	changePicture: PropTypes.func,
	profile: PropTypes.object,
};

export default AccountSettings;
