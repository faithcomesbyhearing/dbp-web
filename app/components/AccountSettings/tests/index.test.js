import React from 'react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import AccountSettings from '..';

Enzyme.configure({ adapter: new Adapter() });

/* eslint-disable react/prop-types */
jest.mock('react-intl', () => ({
	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>,
	defineMessages: (messages) => messages,
}));
jest.mock('../../PopupMessage', () => function accountSettingsPopupMessageMock({ message }) {
  return <span id={'popup-message'}>{message}</span>;
});
/* eslint-enable react/prop-types */

const logout = jest.fn(() => 'Logging out');
const changePicture = jest.fn(() => 'Changing Picture');
const completeProps = {
	logout,
	changePicture,
	profile: {
		email: 'jesse@dbs.org',
		nickname: 'J',
		name: 'Jesse',
		avatar: '',
		popupOpen: false,
	},
};
const incompleteProps = {
	logout,
	changePicture,
	profile: {
		email: '',
		nickname: '',
		name: '',
		avatar: '',
		popupOpen: false,
	},
};

describe('Name of the group', () => {
	it('should match the old snapshot with a full profile', () => {
		const tree = renderer
			.create(<AccountSettings {...completeProps} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
	it('should match the old snapshot without a full profile', () => {
		const tree = renderer
			.create(<AccountSettings {...incompleteProps} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
	it('should match the old snapshot with the popup open', () => {
		const tree = renderer
			.create(<AccountSettings {...completeProps} popupOpen />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
	it('should use handleEmailChange to handle the changed email', () => {
		const wrapper = Enzyme.mount(<AccountSettings {...completeProps} />);
		const spy = jest.spyOn(wrapper.instance(), 'handleEmailChange');
		const newEmail = 'testemail@change.org';
		wrapper.find('input').simulate('change', { target: { value: newEmail } });
		act(() => {
			wrapper.instance().handleEmailChange({ target: { value: newEmail } });
		});
		expect(spy).toHaveBeenCalled();
		expect(wrapper.state('email')).toEqual(newEmail);
		expect(wrapper.find('input').props().value).toEqual(newEmail);
	});
});
