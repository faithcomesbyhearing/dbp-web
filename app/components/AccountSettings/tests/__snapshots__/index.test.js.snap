// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group should match the old snapshot with a full profile 1`] = `
<div
  className="account-settings"
>
  <div
    className="logout-button"
    onClick={[MockFunction]}
    role="button"
    tabIndex={0}
  >
    <span>
      Logout
    </span>
  </div>
  <section
    className="personal-info"
  >
    <svg
      className="avatar-placeholder"
    >
      <use
        xlinkHref="/svglist.svg#avatar_placeholder"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
    <h3
      className="name"
    >
      J
    </h3>
    <span
      className="name"
    >
      Jesse
    </span>
  </section>
  <div
    className="email-section"
  >
    <span
      className="title"
    >
      e-mail
    </span>
    <span
      className="wrapper"
    >
      <svg
        className="icon"
        height="26px"
        width="26px"
      >
        <use
          xlinkHref="/svglist.svg#e-mail"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <input
        onChange={[Function]}
        placeholder="emailaddress@mail.com"
        value="jesse@dbs.org"
      />
    </span>
  </div>
</div>
`;

exports[`Name of the group should match the old snapshot with the popup open 1`] = `
<div
  className="account-settings"
>
  <div
    className="logout-button"
    onClick={[MockFunction]}
    role="button"
    tabIndex={0}
  >
    <span>
      Logout
    </span>
  </div>
  <section
    className="personal-info"
  >
    <svg
      className="avatar-placeholder"
    >
      <use
        xlinkHref="/svglist.svg#avatar_placeholder"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
    <h3
      className="name"
    >
      J
    </h3>
    <span
      className="name"
    >
      Jesse
    </span>
  </section>
  <div
    className="email-section"
  >
    <span
      className="title"
    >
      e-mail
    </span>
    <span
      className="wrapper"
    >
      <svg
        className="icon"
        height="26px"
        width="26px"
      >
        <use
          xlinkHref="/svglist.svg#e-mail"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <input
        onChange={[Function]}
        placeholder="emailaddress@mail.com"
        value="jesse@dbs.org"
      />
    </span>
  </div>
</div>
`;

exports[`Name of the group should match the old snapshot without a full profile 1`] = `
<div
  className="account-settings"
>
  <div
    className="logout-button"
    onClick={[MockFunction]}
    role="button"
    tabIndex={0}
  >
    <span>
      Logout
    </span>
  </div>
  <section
    className="personal-info"
  >
    <svg
      className="avatar-placeholder"
    >
      <use
        xlinkHref="/svglist.svg#avatar_placeholder"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </section>
  <div
    className="email-section"
  >
    <span
      className="title"
    >
      e-mail
    </span>
    <span
      className="wrapper"
    >
      <svg
        className="icon"
        height="26px"
        width="26px"
      >
        <use
          xlinkHref="/svglist.svg#e-mail"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <input
        onChange={[Function]}
        placeholder="emailaddress@mail.com"
        value=""
      />
    </span>
  </div>
</div>
`;
